src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    [ExportInterceptingPropertyValueProvider(PackageIconPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class PackageIconValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private const string PackageIconPropertyName = "PackageIcon";
+        private const string PackPropertyName = "Pack";
+        private const string PackagePathPropertyName = "PackagePath";
+        private const string NoneItem = "None";
+        private const string TrueValue = "True";
+
+        private readonly IProjectItemProvider _sourceItemsProvider;
+        private readonly UnconfiguredProject _unconfiguredProject;
+
+        [ImportingConstructor]
+        public PackageIconValueProvider(
+            [Import(ExportContractNames.ProjectItemProviders.SourceFiles)] IProjectItemProvider sourceItemsProvider,
+            UnconfiguredProject unconfiguredProject)
+        {
+            _sourceItemsProvider = sourceItemsProvider;
+            _unconfiguredProject = unconfiguredProject;
+        }
+
+        // https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packageicon
+        private static string CreatePackageIcon(string filePath, string? packagePath)
+        {
+            string filename = Path.GetFileName(filePath);
+            // Make a slash-only value into empty string, so it won't get prepended onto the path.
+            packagePath = ("\\".Equals(packagePath, StringComparison.Ordinal) || "/".Equals(packagePath, StringComparison.Ordinal))
+                ? string.Empty : (packagePath ?? string.Empty);
```

Comment:
This took me a while to parse. Would this be clearer?

```suggestion
            if (packagePath is null or "\\" or "/")
            {
                packagePath = "";
            }
```
While the comparison isn't explicit, that doesn't matter for these particular literals.

Reply:
Extracted the or-expression into a method to make it more explicit and easier to read left-to-right. Also, added to the comment above this line.

Reply:
I still find this hard to read. And now we have an extra method, which makes it harder to reason about the type's behaviour as a whole.

If you want to use a function here, at least make it local to the caller.

Reply:
Fixed.